pkg/bff/members.go

```diff
@@ -44,13 +46,18 @@ func ConnectMembers(conf config.DirectoryConfig) (_ members.TRISAMembersClient,
 // GetSummaries makes parallel calls to the members service to get the summary
 // information for both testnet and mainnet. If an endpoint returned an error, then a
 // nil value is returned from this function for that endpoint instead of an error.
-func (s *Server) GetSummaries(ctx context.Context) (testnet *members.SummaryReply, mainnet *members.SummaryReply, err error) {
-	rpc := func(ctx context.Context, client members.TRISAMembersClient, network string) (rep proto.Message, err error) {
-		return client.Summary(ctx, &members.SummaryRequest{})
+func (s *Server) GetSummaries(ctx context.Context, testnetID, mainnetID string) (testnet *members.SummaryReply, mainnet *members.SummaryReply, err error) {
+	// Create the RPCs with the VASP ID parameters
+	makeRPC := func(req *members.SummaryRequest) MembersRPC {
+		return func(ctx context.Context, client members.TRISAMembersClient, network string) (rep proto.Message, err error) {
+			return client.Summary(ctx, req)
+		}
 	}
+	testnetRPC := makeRPC(&members.SummaryRequest{Vasp: testnetID})
+	mainnetRPC := makeRPC(&members.SummaryRequest{Vasp: mainnetID})
 
 	// Perform the parallel requests
-	results, errs := s.ParallelMembersRequests(ctx, rpc, false)
+	results, errs := s.ParallelMembersRequests(ctx, testnetRPC, mainnetRPC, false)
```

Comment:
Originally I had envisioned the `rpc` function switching on the `network` inside of the closure; e.g. something more like:

```go
rpc := func(ctx context.Context, client members.TRISAMembersClient, network string) (rep proto.Message, err error) {
    rep := &members.SummaryRequest{}
    switch network {
    case testnet:
        rep.Vasp: testnetID
    case mainnet:
        rep.Vasp: mainnetID
    default:
        return nil, fmt.Errorf("unknown network %q", network)
    }
    return client.Summary(ctx, req) 
}
```

If you think the factory to create closures is the better way; perhaps we should change all of our `Parallel` helper functions to that style. What do you think?

Reply:
I see, I completely forgot about the network parameter but I think that's less clunky than the factory method for now.

Reply:
Ok, so in the future, we'll stick with the `network` switch rather than a factory method and passing two RPCs into the `Parallel`?